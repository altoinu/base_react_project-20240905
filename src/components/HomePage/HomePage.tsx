type HomePageProps = {
  personName: string;
  secondPersonName?: string;
  /**
   * Callback fired when the button is clicked.
   *
   * @param {string} response response from the button click.
   */
  onButtonClick?: (response: string) => void;
};

const HomePage = ({ personName, onButtonClick, ...props }: HomePageProps) => {
  const handleButtonClick = (response: string) => {
    onButtonClick?.(response);
  };

  return (
    <>
      <h1>
        Hello {personName}
        {props.secondPersonName ? ` and ${props.secondPersonName}` : ""}!
      </h1>
      {onButtonClick ? (
        <>
          <button
            type="button"
            onClick={() => handleButtonClick("First button clicked")}
          >
            Click Me!
          </button>
          <button
            type="button"
            onClick={() => handleButtonClick("Second button clicked")}
          >
            Click Me Too!
          </button>
        </>
      ) : (
        ""
      )}
    </>
  );
};

export default HomePage;
