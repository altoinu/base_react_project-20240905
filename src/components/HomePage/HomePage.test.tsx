import HomePage from "./HomePage";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

describe("<HomePage>", () => {
  it("should have person's name", () => {
    render(<HomePage personName="Kaoru" />);
    const heading = screen.getByRole("heading");
    expect(heading.textContent).toBe("Hello Kaoru!");
  });

  it("should have person and second person's names", () => {
    render(<HomePage personName="John" secondPersonName="Doe" />);
    const heading = screen.getByRole("heading");
    expect(heading.textContent).toBe("Hello John and Doe!");
  });

  it("should have two buttons when onButtonClick is set", () => {
    const handleButtonClick = (response: string) => console.log(response);
    render(
      <HomePage
        personName="John"
        secondPersonName="Doe"
        onButtonClick={handleButtonClick}
      />,
    );

    const button1 = screen.getByRole("button", { name: "Click Me!" });
    expect(button1).toBeInTheDocument();
    const button2 = screen.getByRole("button", { name: "Click Me Too!" });
    expect(button2).toBeInTheDocument();
  });

  it("should call onButtonClick with 'First button clicked' when first button is clicked", async () => {
    const user = userEvent.setup();

    const handleButtonClick = jest.fn();
    render(
      <HomePage
        personName="John"
        secondPersonName="Doe"
        onButtonClick={handleButtonClick}
      />,
    );
    const button1 = screen.getByRole("button", { name: "Click Me!" });

    await user.click(button1);

    expect(handleButtonClick).toHaveBeenCalledWith("First button clicked");
  });

  it("should call onButtonClick with 'Second button clicked' when first button is clicked", async () => {
    const user = userEvent.setup();

    const handleButtonClick = jest.fn();
    render(
      <HomePage
        personName="John"
        secondPersonName="Doe"
        onButtonClick={handleButtonClick}
      />,
    );
    const button2 = screen.getByRole("button", { name: "Click Me Too!" });

    await user.click(button2);

    expect(handleButtonClick).toHaveBeenCalledWith("Second button clicked");
  });
});
