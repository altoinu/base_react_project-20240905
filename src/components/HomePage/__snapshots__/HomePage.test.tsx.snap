// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage> should render 1`] = `
<div>
  <h1>
    Hello 
    John
     and Doe
    !
  </h1>
  <div>
    <button
      type="button"
    >
      Click Me!
    </button>
    <button
      type="button"
    >
      Click Me Too!
    </button>
  </div>
</div>
`;
